benchmarks/solubility/plugin/solubility.cc

```diff
@@ -0,0 +1,409 @@
+/*
+  Copyright (C) 2014 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/melt.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/interface.h>
+
+#include <array>
+#include <utility>
+#include <limits>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+     * Volatiles material model.
```

Comment:
The way I envisioned this is that what I wrote here is really only the benchmark. If @danieldouglas92 wants to implement a more complex model, I thought he would just make a copy of this, because I was assuming that the two of you might want to change a lot of things and this is really just a minimal working example. 

You could then pick a more general name for your material model.